### PR TITLE
Modify verifyData receives packets instead of result data

### LINKED ISSUE #1405: Modify verifyData receives packets instead of result data
- [ ] function to get hash of requestPacket (encode struct to protobuf style)
- [ ] function to get hash of responsePacket (encode struct to protobuf style)
- [ ] Append 2 hash and hash to get resultHash
- [ ] Modify verifyData to use result hash

*Ref
https://developers.google.com/protocol-buffers/docs/encoding

### PR DESCRIPTION
**Pull request by taobun**
fixed #1405

### CODE REVIEW

File: bridges/evm/contracts/Bridge.sol [hunk 1]
```diff
@@ -135,66 +136,16 @@ contract Bridge is IBridge, Ownable {
...
-
-    //     return result;
-    // }
-
     /// Verifies that the given data is a valid data on BandChain as of the given block height.
     /// @param _blockHeight The block height. Someone must already relay this block.
-    /// @param _data The data to verify, with the format similar to what on the blockchain store.
-    /// @param _requestId The ID of request for this data piece.
+    /// @param _requestPacket The request packet contain detail of request.
+    /// @param _responsePacket The response packet of this request contain metadata of result.
```
Comment (sorawit) [inline bridges/evm/contracts/Bridge.sol]: Let's remove "contain...." part. Just say The request packet is this request, The response packet of this request.

File: bridges/evm/contracts/Packets.sol [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    function marshalRequestPacket(IBridge.RequestPacket memory _self)
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return
+            abi.encodePacked(
+                hex"d9c58927",
+                uint8(10), // (1 << 3) | 2
+                Utils.encodeVarintUnsigned(bytes(_self.clientId).length),
```
Comment (sorawit) [inline bridges/evm/contracts/Packets.sol]: Why don't we use `getEncodeLength`?

File: bridges/evm/contracts/Packets.sol [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    function marshalResponsePacket(IBridge.ResponsePacket memory _self)
+        internal
+        pure
+        returns (bytes memory)
+    {
+        return
+            abi.encodePacked(
+                hex"79b5957c",
```
Comment (sorawit) [inline bridges/evm/contracts/Packets.sol]: Explain it

**Review by sorawit — CHANGES_REQUESTED** [review]

File: bridges/evm/contracts/Packets.sol [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        return
+            abi.encodePacked(
+                hex"d9c58927", // Amino codec info for RequestPacket type
+                uint8(10), // (1 << 3) | 2
+                getEncodeLength(_self.clientId),
+                _self.clientId,
+                uint8(16), // (2 << 3) | 0
+                Utils.encodeVarintUnsigned(_self.oracleScriptId),
+                uint8(26), // (3 << 3) | 2
+                Utils.encodeVarintUnsigned(bytes(_self.params).length),
```
Comment (sorawit) [inline bridges/evm/contracts/Packets.sol]: and here?

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by Benzbeeb — APPROVED** [review]

**Pull request merged**